### DISCUSSION TITLE
More candidates for toolchains on linux

### DISCUSSION THREAD

**Discussion by jhheider** [pkgxdev/brewkit#304] [Ideas]
- gnu.org/autoconf
- gnu.org/automake
- gnu.org/libtool
- (possibly) perl.org

lots of tools (cough *configure* cough) expect some or all of these as a matter of course. the most recent yarnpkg.com build failure was buried in a log where the `mozjpeg` module was trying to recompile its binary bits.

obvious downside is masking a runtime dependency (mostly only in the case of `perl`.

note that unless explicitly adding `pkg-config`, `automake` doesn't seem to be able to find `pkg.m4`.

**Comment by mxcl** [comment]
> obvious downside is masking a runtime dependency (mostly only in the case of perl.

the tests don’t get the toolchain so thorough tests should be enough

**Reply by jhheider** [reply]
it's a great point.

**Comment by mxcl** [comment]
> note that unless explicitly adding pkg-config, automake doesn't seem to be able to find pkg.m4.

ah yes. This would indeed be the danger of shims. We could have brewkit always add pkg-config to the env. Dunno. Let's decide next time a build error like that happens.

**Reply by jhheider** [reply]
that's a deep cut, certainly. i expect that to be rarely found.
